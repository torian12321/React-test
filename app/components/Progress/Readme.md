Basic
```
<div>
	<Progress />
	<Progress val={80} />
	<Progress val={20} />
</div>
```

States
```
initialState = {val: 50};

<div>
	<Range 
		value    = {state.val}
		onChange = {(e) => setState({val: e})}
		texted
	/>

	<Progress val={state.val} state='main'/>
	<Progress val={state.val} state='sec'/>
	<Progress val={state.val} state='success'/>
	<Progress val={state.val} state='danger'/>
	<Progress val={state.val} state='warning'/>
</div>
```

Sizes
```
initialState = {val: 50};

<div>
	<Range 
		value    = {state.val}
		onChange = {(e) => setState({val: e})}
		texted
	/>

	<Progress val={state.val} size='xs'/>
	<Progress val={state.val} size='sm'/>
	<Progress val={state.val} size='md'/>
	<Progress val={state.val} size='lg'/>
</div>
```

------------
Progress - Circular
------------
Check the reference [here](https://codepen.io/mavrK/pen/pRGPNO?editors=1000)

Basic
```
<Progress type='circular'/>
```

States
```
initialState = {val: 50};

<div>
	<Range 
		value    = {state.val}
		onChange = {(e) => setState({val: e})}
		texted
	/>

	<Progress val={state.val} type='circular' state='main'/>
	<Progress val={state.val} type='circular' state='sec'/>
	<Progress val={state.val} type='circular' state='success'/>
	<Progress val={state.val} type='circular' state='danger'/>
	<Progress val={state.val} type='circular' state='warning'/>
</div>
```

Sizes
```
<div>
	<Progress type='circular' size='xs'/>
	<Progress type='circular' size='sm'/>
	<Progress type='circular' size='md'/>
	<Progress type='circular' size='lg'/>
</div>
```