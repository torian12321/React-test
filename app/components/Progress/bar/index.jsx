var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames'),
styles    = require('./_style.less');

const ProgressBar = (props) =>(
    <div className={props.className}>
        <div style={{ width: `${props.val}%` }}>
            {`${props.val}%`}
        </div>
    </div>
);


ProgressBar.propTypes = {
    val      : PropTypes.number.isRequired,
    className: PropTypes.string
};

module.exports = ProgressBar;
