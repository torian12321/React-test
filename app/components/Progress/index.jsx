var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames'),
Bar       = require('./bar'),
Circle    = require('./circular');

const Progress = props =>{
    let conf = {
        val      : props.val,
        className: classNames(
            'pb',
            `pb-${props.type}`,
            props.state,
            props.size,
            props.className
        )
    };


    if(props.type === 'circular'){
        return <Circle {...conf} />
    }else{
        return <Bar {...conf} />
    }
};


Progress.propTypes = {
    type : PropTypes.oneOf(['bar', 'circular']),
    val  : PropTypes.number,
    label: PropTypes.string,    
    size : PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
    state: PropTypes.oneOf(['main', 'sec', 'info', 'success', 'danger', 'warning'])
};
Progress.defaultProps = {
    type: 'bar',
    val : 50
};

module.exports = Progress;