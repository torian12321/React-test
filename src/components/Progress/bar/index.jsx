import React      from 'react';
import PropTypes  from 'prop-types';
import './_style';

class ProgressBar extends React.PureComponent {
  render(){
    return(
      <div className={this.props.className}>
        <div style={{ width: `${this.props.val}%` }}>
          {`${this.props.val}%`}
        </div>
      </div>
    );
  }
}

ProgressBar.propTypes = {
  val      : PropTypes.number.isRequired,
  className: PropTypes.string
};

module.exports = ProgressBar;
