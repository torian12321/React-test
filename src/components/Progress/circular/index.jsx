import React from 'react';
import PropTypes from 'prop-types';
import styles from './_style';

const ProgressCircle = (props) =>{
    const offset = (( -parseFloat(props.val) /100) * 220) - 220;

    return(
    	<div className={props.className}>
           <label>{`${props.val}%`}</label>
            <svg viewBox="0 0 80 80">
                <circle cx="40" cy="40" r="35" />
                <circle cx="40" cy="40" r="35" className="fill" strokeDashoffset={offset} />
            </svg>
        </div>
    );
};


ProgressCircle.propTypes = {
    val      : PropTypes.number.isRequired,
    className: PropTypes.string
};

module.exports = ProgressCircle;
