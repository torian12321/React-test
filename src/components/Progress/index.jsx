import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import Bar        from './bar';
import Circle     from './circular';

class Progress extends React.PureComponent {
  render(){
    let conf = {
      val      : this.props.val,
      className: classNames(
        'pb',
        `pb-${this.props.type}`,
        this.props.state,
        this.props.size,
        this.props.className
      )
    };

    if(this.props.type === 'circular'){
      return <Circle {...conf} />;
    }else{
      return <Bar {...conf} />;
    }
  }
}


Progress.propTypes = {
  type : PropTypes.oneOf(['bar', 'circular']),
  val  : PropTypes.number,
  label: PropTypes.string,    
  size : PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
  state: PropTypes.oneOf(['main', 'sec', 'info', 'success', 'danger', 'warning'])
};
Progress.defaultProps = {
  type: 'bar',
  val : 50
};

module.exports = Progress;