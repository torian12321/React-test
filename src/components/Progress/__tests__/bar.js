import React    from 'react';
import renderer from 'react-test-renderer';
import Progress from '../index';

describe('component', () => {
  describe('<Progress />', () => {
    describe('Bar', () => {
    
      it('should render correctly', () => {
        expect(renderer.create(
          <Progress />
        )).toMatchSnapshot();
      });

      it('should render 73 progress', () => {
        expect(renderer.create(
          <Progress val={73} />
        )).toMatchSnapshot();
      });

      it('should render small', () => {
        expect(renderer.create(
          <Progress size='sm' />
        )).toMatchSnapshot();
      });

      it('should render info state', () => {
        expect(renderer.create(
          <Progress state='info' />
        )).toMatchSnapshot();
      });

    });
  });
});
