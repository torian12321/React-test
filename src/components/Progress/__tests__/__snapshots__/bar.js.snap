// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Progress /> Bar should render 73 progress 1`] = `
<div
  className="pb pb-bar"
>
  <div
    style={
      Object {
        "width": "73%",
      }
    }
  >
    73%
  </div>
</div>
`;

exports[`component <Progress /> Bar should render correctly 1`] = `
<div
  className="pb pb-bar"
>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;

exports[`component <Progress /> Bar should render info state 1`] = `
<div
  className="pb pb-bar info"
>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;

exports[`component <Progress /> Bar should render small 1`] = `
<div
  className="pb pb-bar sm"
>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
  >
    50%
  </div>
</div>
`;
